packages/wallet/src/services/TransactionReemitter.ts

```diff
@@ -0,0 +1,81 @@
+import { Cardano } from '@cardano-sdk/core';
+import { DocumentStore } from '../persistence';
+import { Logger, dummyLogger } from 'ts-log';
+import { NewTxAlonzoWithSlot } from './types';
+import { Observable, filter, from, map, merge, mergeMap, scan, tap, withLatestFrom } from 'rxjs';
+
+interface TransactionReemitterProps {
+  rollback$: Observable<Cardano.TxAlonzo>;
+  confirmed$: Observable<NewTxAlonzoWithSlot>;
+  submitting$: Observable<Cardano.NewTxAlonzo>;
+  store: DocumentStore<NewTxAlonzoWithSlot[]>;
+  stabilityWindowSlotsCount?: number;
+  logger?: Logger;
+}
+
+enum txSource {
+  store,
+  confirmed,
+  submitting
+}
+
+// 3k/f (where k is the security parameter in genesis, and f is the active slot co-efficient parameter
+// in genesis that determines the probability for amount of blocks created in an epoch.)
+const kStabilityWindowSlotsCount = 129_600; // 3k/f on current mainnet
+
+export const createTransactionReemitter = ({
+  rollback$,
+  confirmed$,
+  submitting$,
+  store,
+  stabilityWindowSlotsCount = kStabilityWindowSlotsCount,
+  logger = dummyLogger
+}: TransactionReemitterProps): Observable<Cardano.NewTxAlonzo> => {
+  const volatileTransactions$ = merge(
+    store.get().pipe(
+      mergeMap((txs) => from(txs)),
+      map((tx) => ({ source: txSource.store, tx }))
+    ),
+    confirmed$.pipe(map((tx) => ({ source: txSource.confirmed, tx }))),
+    submitting$.pipe(map((tx) => ({ source: txSource.submitting, tx: { ...tx, slot: null! } })))
+  ).pipe(
+    scan((volatiles, { tx, source }) => {
+      switch (source) {
+        case txSource.store: {
+          // Do not calculate stability window for old transactions coming from the store
+          volatiles = [...volatiles, tx];
+          break;
+        }
+        case txSource.submitting: {
+          // Transactions in submitting are the ones reemitted. Remove them from volatiles
+          volatiles = volatiles.filter((v) => v.id !== tx.id);
+          store.set(volatiles);
+          break;
+        }
+        case txSource.confirmed: {
+          const oldestAcceptedSlot = tx.slot > stabilityWindowSlotsCount ? tx.slot - stabilityWindowSlotsCount : 0;
+          // Remove transactions considered stable
+          volatiles = [...volatiles.filter(({ slot }) => slot > oldestAcceptedSlot), tx];
```

Comment:
Might as well filter out transactions where slot is > of [validityInterval](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Cardano/types/Transaction.ts#L42), cause those will fail resubmission anyway

Reply:
Fixed here [19b46b1](https://github.com/input-output-hk/cardano-js-sdk/pull/351/commits/19b46b15c9ebd6072e2e4e00645e24a10b648e45)